nestkernel/simulation_manager.cpp

```diff
@@ -102,6 +102,7 @@ nest::SimulationManager::set_status( const DictionaryDatum& d )
   TimeConverter time_converter;
 
   double time;
+  const double RESOLUTION_TOLERANCE = 1e-14;
```

Comment:
Why can't [`std::numeric_limits<T>::epsilon`](https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon) be used here? I dislike the idea of hard-coded magic number (especially if they come without any comment)?

Reply:
Good point. I tried, but I found the error is actually much larger than than this epsilon (2.2e-16).

For example if I try to set the resolution to `0.102` ms and leave the `tics_per_ms` at 1000, it should work, but there is a significant rounding error happening.

`resd / Time::get_ms_per_tic()` should yield a round `102`, but is actually `101.99999999999998578915`.

Reply:
Ah, it is all relative ... For a value around 100, round-off error should be around 100 * eps ~ 10^-14, as is the case here. So I'd suggest

```C++
bool is_integer( double n )
{
  double int_part;
  double frac_part = std::modf( n, &int_part );

  // Since n > 0 and modf always rounds towards zero, a value of n just below an 
  // integer will result in frac_part = 0.99999.... . We subtract from 1 in this case. 
  if ( frac_part > 0.5 )
  {
    frac_part = 1 - frac_part;
  }
  
  // factor 4 allows for two bits of rounding error
  return frac_part < 4 * n * std::numeric_limits<double>::epsilon;
}
```

This should probably go somewhere in `libnestutil`.

Reply:
Alright, that's a good solution. I hope that's better now.

Reply:
Awesome!